WordPress/gutenberg
# WordPressComponentProps: Only add refs when they are actually added

## What?

Remove the `ref` prop from `WordPressComponentProps`.

The ad hoc `Omit`s on the following components have been removed:

- CheckboxControl
- Disabled
- FormToggle
- Placeholder
- RadioControl
- TextareaControl

## Why?

The `WordPressComponentProps` type always added a `ref` prop, even though the `props` object of a `ForwardRefRenderFunction` (i.e. the function to be passed to `forwardRef()`) should never even contain a `ref` prop.

The `ref` prop should only be mixed into the `props` object type only after the component is wrapped by `forwardRef()`.

## How?

Don't include the `ref` in `WordPressComponentProps`, but instead allow it to be added from `WordPressComponentFromProps` or `WordPressComponent`.

## Testing Instructions

1. `npm run storybook:dev`
2. In the Docs view for the affected components listed above, see that the `ref` prop is still omitted from the props table.
